src/bin/copy-int-vector.cc

```diff
@@ -22,6 +22,28 @@
 #include "matrix/kaldi-vector.h"
 #include "transform/transform-common.h"
 
+namespace kaldi {
+
+// Converts a integer vector into a list of (integer, length) pairs
+// e.g. 0 0 1 1 1 1 1 10 10 10 --> 0 2 ; 1 5 ; 10 3 ;
+void VectorToLengthsPair(const std::vector<int32> &vec, 
+                         std::vector<std::pair<int32, int32> > *pairs) {
+  int32 prev_val = -1, prev_count = 0;
```

Comment:
Common name for this technique is "run length encoding" (RLE), meaning that each run of consecutive identical elements is encoded with its value and length.

The switch name `--write-lengths` is confising, I would rather go with `--write-rle` or `--write-length-encoded`.

I am worried that the vector written in this format cannot be transformed back. I think we need to introduce a complimentary pair of switches to support format conversion, `--read-rle` and `--write-rle`, in this utility.